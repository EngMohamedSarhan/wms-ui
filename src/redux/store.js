import { persistStore } from "redux-persist";

import { configureStore } from "@reduxjs/toolkit";

import { persistedReducer } from "./reducers/reducers";

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export const persistor = persistStore(store);

export const settingsState = (state) => state.settingsReducer;
