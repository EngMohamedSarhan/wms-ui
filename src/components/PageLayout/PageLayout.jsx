import PropTypes from "prop-types";
import React, { useEffect } from "react";

import NotificationModal from "../../components/NotificationModal/NotificationModal";
import AppLoader from "../AppLoader/AppLoader";
import NavBar from "../NavBar/NavBar";
import { useSelector } from "react-redux";
import { settingsState } from "../../redux/store";
import useGetWarehouseLocations from "../../hooks/useGetWarehouseLocations";
import ErroryBoundary from "../ErrorBoundary/ErroryBoundary";

const PageLayout = (props) => {
  const { warehouses } = useSelector(settingsState);
  const { getWarehouseLocations } = useGetWarehouseLocations();

  useEffect(() => {
    if (warehouses.length === 0) {
      getWarehouseLocations();
    }
  }, []);
  return (
    <ErroryBoundary>
      <div>
        <NavBar />
        <AppLoader />
        <NotificationModal />
        <main {...props} />
      </div>
    </ErroryBoundary>
  );
};

PageLayout.propTypes = {
  children: PropTypes.node,
};

export default PageLayout;
