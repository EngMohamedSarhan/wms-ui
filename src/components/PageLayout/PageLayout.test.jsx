/* eslint-disable no-undef */
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React from "react";
import { Provider } from "react-redux";
import { Router } from "react-router";

import { store } from "../../redux/store";
import PageLayout from "./PageLayout";

describe("PageLayout Component", () =>
  test("PageLayout render", () => {
    const history = createMemoryHistory({ initialEntries: ["/"] });
    const { container, queryByTestId, getByAltText } = render(
      <Provider store={store}>
        <Router location={history.location} navigator={history}>
          <PageLayout />
        </Router>
      </Provider>
    );
    const loader = queryByTestId("application-loader");
    const modal = queryByTestId("notification-modal");
    const logo = getByAltText("logo");
    const header = container.getElementsByTagName("h2")[0];
    expect(header).toBeInTheDocument();
    expect(logo).toBeInTheDocument();
    expect(loader).toBeInTheDocument();
    expect(modal).toBeNull();
  }));
