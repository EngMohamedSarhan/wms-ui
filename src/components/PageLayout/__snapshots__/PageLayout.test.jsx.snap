// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout Component PageLayout snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "getServerState": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Router
    location={
      Object {
        "key": "1",
      }
    }
  >
    <PageLayout />
  </Router>
</Context.Provider>
`;
